### PR TITLE
updated hashing method and seed value to align with allocation calculation on other platforms

### PR DESCRIPTION
**Pull request by jasonwaters**
## Description

Updated hashing method and seed value to align with allocation calculation on other platforms.

- Switched to use the guava library and use the hashUnencodedChars method to compute allocation.
- Updated the salt value used in the deviceId to "0".
- Updated tests

## Related Issue

https://jira.corp.adobe.com/browse/TNT-40082

## How Has This Been Tested?

- Added `AllocationUtilsTest`
- I also updated existing tests, where new allocation values caused a different experience to be selected.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -134,6 +134,7 @@ dependencies {
     implementation "com.fasterxml.jackson.core:jackson-databind:2.11.3"
     implementation "com.konghq:unirest-java:3.11.02"
     implementation "io.github.jamsesso:json-logic-java:1.0.5"
```
Comment (artur-ciocanu) [inline build.gradle:136]: Could we avoid adding Guava as a dependency. Guava is used in a lot of projects and we might force customer into a lot of pain when dealing with this dependency. We have experienced this when using Hadoop, Cassandra etc

**Review by artur-ciocanu — CHANGES_REQUESTED** [review]
@jasonwaters I added a comment, I think it is better to have the hashing method as standalone without using Guava. As a general rule we should try to keep our dependencies to a minimum.

File: build.gradle (hunk 1 above)
Comment (jasonwaters) [reply build.gradle]: @artur-ciocanu Thanks for the feedback.  I have removed the dependency on guava.

**Comment by jasonwaters** [comment]
@artur-ciocanu, @5amfung I can haz approve?

**Review by 5amfung — APPROVED** [review]

**Pull request merged**